src/storage-client/src/controller.rs

```diff
@@ -420,6 +424,33 @@ pub enum ReadPolicy<T> {
     Multiple(Vec<ReadPolicy<T>>),
 }
 
+impl<T> ReadPolicy<T>
+where
+    T: Timestamp + TimestampManipulation,
+{
+    /// Creates a read policy that lags the write frontier "by one".
+    pub fn step_back() -> Self {
+        Self::LagWriteFrontier(Arc::new(move |upper| {
```

Comment:
What was the method we landed on in the end, again?

Reply:
You have two variables `since` and `next_since` as your state. The initial value for both is the current since of the collection. 

For each observed `upper` if `next_since.iter().all(|t| !upper.less_equal(t))) == true`:
    1. `since = next_since`
    2. `next_since = upper`

In all cases you return `since` as the result of the policy